x86.c

```diff
@@ -670,3 +685,112 @@ void os_close() {
 }
 
 #endif
+
+// For 64 bit macOS.
+#ifdef mac_os
+
+void os_getchar() {
```

Comment:
I think we should add a comment explaining that MacOS follows the System V AMD64 ABI or try to combine these functions with the ones used with `x86_64` because they are so similar.

Reply:
Maybe we could have a block that defines the syscall numbers (because those vary between OS, and are not specified by the SystemV ABI), and then a block
Something like:

```
#ifdef mac
#define ABI_SYSTEM_V

#define OS_GETCHAR_SYSCALL 0x2000003
#define OS_EXIT_SYSCALL ...
#endif

#ifdef ABI_SYSTEM_V

void os_getchar() {
 int lbl = alloc_label();
  mov_reg_imm(AX, OS_GETCHAR_SYSCALL);    // mov  rax, 
  push_reg(AX);          // push rax          # buffer to read byte
  mov_reg_imm(DI, 0);    // mov  rdi, 0       # rdi = 0 = STDIN
  mov_reg_imm(DX, 1);    // mov  rdx, 1       # rdx = 1 = number of bytes to read
  mov_reg_reg(SI, SP);   // mov  rsi, rsp     # to the stack
  syscall();             // syscall
  xor_reg_reg(DX, DX);   // xor  rdx, rdx     # rdx = 0
  cmp_reg_reg(AX, DX);   // cmp  rax, rdx
  pop_reg(AX);           // pop  rax
  jump_cond(NE, lbl);    // jne  lbl          # if byte was read don't return EOF
  mov_reg_imm(AX, -1);   // mov  rax, -1      # -1 on EOF
  def_label(lbl);        // end label
}
```